### PR TITLE
ios: contact aliases

### PR DESCRIPTION
**Pull request by jr-simplex**

### CODE REVIEW

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 1]
```diff
@@ -598,7 +635,11 @@ public struct GroupMember: Identifiable, Decodable {
     public var chatViewName: String {
         get {
             let p = memberProfile
-            return p.displayName + (p.fullName == "" || p.fullName == p.displayName ? "" : " / \(p.fullName)")
+            if p.localAlias == "" {
+                return p.displayName + (p.fullName == "" || p.fullName == p.displayName ? "" : " / \(p.fullName)")
+            } else {
+                return p.localAlias
+            }
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/ChatTypes.swift:642]:
```suggestion
            return p.localAlias == ""
                        ? p.displayName + (p.fullName == "" || p.fullName == p.displayName ? "" : " / \(p.fullName)")
                        : p.localAlias
```

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 2]
```diff
@@ -97,11 +106,18 @@ public protocol NamedChat {
...
 
 extension NamedChat {
     public var chatViewName: String {
-        get { displayName + (fullName == "" || fullName == displayName ? "" : " / \(fullName)") }
+        get {
+            if localAlias == "" {
+                return displayName + (fullName == "" || fullName == displayName ? "" : " / \(fullName)")
+            } else {
+                return localAlias
+            }
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/ChatTypes.swift:119]:
```suggestion
            localAlias == ""
                ? displayName + (fullName == "" || fullName == displayName ? "" : " / \(fullName)")
                : localAlias
```

File: apps/ios/SimpleXChat/ChatTypes.swift (hunk 1 above)
Comment (epoberezkin) [reply apps/ios/SimpleXChat/ChatTypes.swift:642]: why it's not using chatViewName here?

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 3]
```diff
@@ -524,6 +554,7 @@ public struct GroupInfo: Identifiable, Decodable, NamedChat {
     public var displayName: String { get { groupProfile.displayName } }
     public var fullName: String { get { groupProfile.fullName } }
     public var image: String? { get { groupProfile.image } }
+    public var localAlias: String { get { "" } }
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/ChatTypes.swift:557]:
```suggestion
    public var localAlias: String { "" }
```

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 4]
```diff
@@ -559,6 +590,7 @@ public struct GroupProfile: Codable, NamedChat {
     public var displayName: String
     public var fullName: String
     public var image: String?
+    public var localAlias: String { get { "" } }
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/ChatTypes.swift:593]:
```suggestion
    public var localAlias: String { "" }
```

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 5]
```diff
@@ -425,6 +454,7 @@ public struct PendingContactConnection: Decodable, NamedChat {
     }
     public var fullName: String { get { "" } }
     public var image: String? { get { nil } }
+    public var localAlias: String { get { "" } }
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/ChatTypes.swift:457]:
```suggestion
    public var localAlias: String { "" }
```

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 6]
```diff
@@ -384,6 +412,7 @@ public struct UserContactRequest: Decodable, NamedChat {
     public var displayName: String { get { profile.displayName } }
     public var fullName: String { get { profile.fullName } }
     public var image: String? { get { profile.image } }
+    public var localAlias: String { get { "" } }
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/ChatTypes.swift:415]:
```suggestion
    public var localAlias: String { "" }
```

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 7]
```diff
@@ -303,9 +330,10 @@ public struct Contact: Identifiable, Decodable, NamedChat {
     public var apiId: Int64 { get { contactId } }
     public var ready: Bool { get { activeConn.connStatus == .ready } }
     public var sendMsgEnabled: Bool { get { true } }
-    public var displayName: String { get { profile.displayName } }
+    public var displayName: String { get { localAlias == "" ? profile.displayName : localAlias } }
     public var fullName: String { get { profile.fullName } }
     public var image: String? { get { profile.image } }
+    public var localAlias: String { get { profile.localAlias } }
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/ChatTypes.swift:336]:
```suggestion
    public var localAlias: String { profile.localAlias }
```

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 8]
```diff
@@ -303,9 +330,10 @@ public struct Contact: Identifiable, Decodable, NamedChat {
     public var apiId: Int64 { get { contactId } }
     public var ready: Bool { get { activeConn.connStatus == .ready } }
     public var sendMsgEnabled: Bool { get { true } }
-    public var displayName: String { get { profile.displayName } }
+    public var displayName: String { get { localAlias == "" ? profile.displayName : localAlias } }
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/ChatTypes.swift:333]:
```suggestion
    public var displayName: String { localAlias == "" ? profile.displayName : localAlias }
```

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 9]
```diff
@@ -97,11 +106,18 @@ public protocol NamedChat {
     var displayName: String { get }
     var fullName: String { get }
     var image: String? { get }
+    var localAlias: String { get }
 }
 
 extension NamedChat {
     public var chatViewName: String {
-        get { displayName + (fullName == "" || fullName == displayName ? "" : " / \(fullName)") }
+        get {
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/ChatTypes.swift:114]: no need for explicit gets

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 10]
```diff
@@ -55,31 +57,38 @@ public struct Profile: Codable, NamedChat {
...
     public var image: String?
+    public var localAlias: String
 
     var profileViewName: String {
-        (fullName == "" || displayName == fullName) ? displayName : "\(displayName) (\(fullName))"
+        if localAlias == "" {
+            return (fullName == "" || displayName == fullName) ? displayName : "\(displayName) (\(fullName))"
+        } else {
+            return localAlias
+        }
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/ChatTypes.swift:79]: also chatViewName? or ternary

File: apps/ios/Shared/Views/ChatList/ChatListView.swift [hunk 11]
```diff
@@ -77,9 +77,15 @@ struct ChatListView: View {
...
-            : chatModel.chats.filter {
-                $0.chatInfo.chatType != .contactConnection &&
-                $0.chatInfo.chatViewName.localizedLowercase.contains(s)
+            : chatModel.chats.filter { chat in
+                switch chat.chatInfo {
+                case let .direct(contact):
+                    return contact.chatViewName.localizedLowercase.contains(s)
+                    || contact.profile.displayName.localizedLowercase.contains(s)
+                    || contact.fullName.localizedLowercase.contains(s)
+                case .contactConnection: return false
```
Comment (epoberezkin) [inline apps/ios/Shared/Views/ChatList/ChatListView.swift:86]: why is it filtered out?

File: apps/ios/SimpleXChat/ChatTypes.swift (hunk 1 above)
Comment (jr-simplex) [reply apps/ios/SimpleXChat/ChatTypes.swift:642]: if you mean profileViewName it's different

**Pull request merged**